applications/KWS_Phoneme/kwscnn.py

```diff
@@ -0,0 +1,677 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT license.
+
+import torch
+import torch.nn as nn
+from torch.autograd import Variable
+import torch.nn.functional as F
+from torch.nn.parameter import Parameter
+from utils import _single
+import math
+import numpy as np 
+from edgeml_pytorch.graph.rnn import *
+
+class _IndexSelect(torch.nn.Module):
+    def __init__(self, channels, direction, groups):
+        """
+        Channel permutation module. The purpose of this is to allow mixing across the CNN groups
```

Comment:
If you're using full-stops once in a statement, please use them throughout that statement.